src/main/java/model/canvas/Canvas.java

```diff
@@ -12,14 +13,67 @@
 
   private final CanvasUpdateListenerComposite canvasUpdateListeners;
   private final List<ILayer> layers;
-  private ILayer activeLayer; // TODO use when the ILayer interface has been further specified
+  private ILayer activeLayer;
 
   public Canvas() {
     canvasUpdateListeners = new CanvasUpdateListenerComposite();
     layers = new ArrayList<>(20);
     activeLayer = null;
   }
 
+  /**
+   * Verifies that there is an active layer. If that isn't the case, an exception is thrown.
+   *
+   * @throws IllegalStateException if there is no active layer.
+   */
+  private void verifyActiveLayerExistence() {
+    if (activeLayer == null) {
+      throw new IllegalStateException("No available active layer!");
+    }
+  }
+
+  /**
+   * Sets the color of the pixel at the specified coordinate, in the active layer.
+   *
+   * @param x the x-coordinate of the pixel that will be changed.
+   * @param y the y-coordinate of the pixel that will be changed.
+   * @param color the new color of the pixel.
+   * @throws IllegalStateException if there is no active layer.
+   * @throws IndexOutOfBoundsException if the specified coordinate is out-of-bounds.
+   * @throws NullPointerException if any arguments are {@code null}.
+   */
```

Comment:
Could be valuable to specify that the origin is at the top left corner and that the coordinates are zero-indexed.